privacyidea/lib/eventhandler/usernotification.py

```diff
@@ -311,15 +315,36 @@ def do(self, action, options=None):
                 ua_browser=request.user_agent.browser,
                 ua_string=request.user_agent.string
             )
+            subject = subject.format(
+                admin=logged_in_user.get("username"),
+                realm=logged_in_user.get("realm"),
+                action=request.path,
+                serial=serial,
+                url=request.url_root,
+                user=tokenowner.info.get("givenname"),
+                surname=tokenowner.info.get("surname"),
+                givenname=recipient.get("givenname"),
+                username=tokenowner.login,
+                userrealm=tokenowner.realm,
+                tokentype=tokentype,
+                registrationcode=registrationcode,
+                recipient_givenname=recipient.get("givenname"),
+                recipient_surname=recipient.get("surname"),
+                googleurl_value=googleurl_value,
+                time=time,
+                date=date,
+                client_ip=g.client_ip,
+                ua_browser=request.user_agent.browser,
+                ua_string=request.user_agent.string
+            )
```

Comment:
Hm, this is pretty much a duplicate of the text format code above. Maybe we can avoid the duplication by using a dictionary, or so?
```python
values = dict(admin=..., )
text = text.format(googleurl_img=googleimg_url, **values)
subject = subject.format(**values)
```

Reply:
done.